
## Dataset taxonomy

table of genomes and taxonomy used in this study [organism_taxonomy.xls](https://github.com/Joseph7e/Marker-genes-as-predictors-of-shared-genomic-function/blob/master/Data/organism_taxonomy.xls)


## Gene Extracting

log of 16S sequences extracted from genomes is located in [gene_and_length_all.xls)](https://github.com/Joseph7e/Marker-genes-as-predictors-of-shared-genomic-function/blob/master/Data/gene_and_length_all.xls).



log of variable regions successfully extracted using amplify_fasta.py is found in[amplified_16S_summary.xls](https://github.com/Joseph7e/Marker-genes-as-predictors-of-shared-genomic-function/blob/master/Data/amplified_16S_summary.xls)


## intra organism variation statistics

A table with  the total 16S gene copies, sequence lengths (min, max, and average), minimum percent identity, and average percent identity per genome for the complete 16S genes [complete_variation_table.xls](https://github.com/Joseph7e/Marker-genes-as-predictors-of-shared-genomic-function/blob/master/Data/complete_variation_table.xls)  and variable portion of the gene [amplified_variation_table.xls](https://github.com/Joseph7e/Marker-genes-as-predictors-of-shared-genomic-function/blob/master/Data/amplified_variation_table.xls) 


## Orthofinder

A complete list of pairwise comparisons for the 16S variable region dataset [amplified_16S_comparisons.xls](https://github.com/Joseph7e/Marker-genes-as-predictors-of-shared-genomic-function/blob/master/Data/amplified_16S_comparisons.xls)



## clustering sequences

The data sheet [all_unaligned_clustalInfo.csv](https://github.com/Joseph7e/Marker-genes-as-predictors-of-shared-genomic-function/blob/master/Data/all_unaligned_clustalInfo.csv) containes the parsed uclust clustering data for each clustering group.

The data sheet [all_unaligned_clustalSummary.xls](https://github.com/Joseph7e/Marker-genes-as-predictors-of-shared-genomic-function/blob/master/Data/all_unaligned_clustalSummary.xls) contains a summary of 
